File: 217-contains-duplicate/217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
//     let hashmap = {}
    
//     for (let values of nums) {
//         if (!hashmap[values]) {
//             hashmap[values] = 1
//         } else {
//             hashmap[values]++
//         }
//     }
    
//     for (let values in hashmap) {
//         if (hashmap[values] > 1) {
//             return true
//         } 
//     }
//     return false
    
    let set = new Set()
    for (let num of nums) {
        if(set.has(num)) {
            return true
        }
                set.add(num)

    }
    return false
};

File: 217-contains-duplicate/217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        counter = {}
        for num in nums:
            if num not in counter:
                counter[num] = 0
            counter[num] += 1
        for num, freq in counter.items():
            if freq > 1:
                return True
        return False
        
        
        
        
        
        
